File: js/s6-optional-challenges/are-there-duplicates-multiple-pointers.js
function areThereDuplicates(...args) {
    args.sort();

    let left = 0;
    let right = 1;

    while (right < args.length) {
        if (args[left] === args[right]) {
            return true;
        }

        left++;
        right++;
    }

    return false;
}

console.log(areThereDuplicates());

File: dart/s6_optional_challanges/are_there_duplicates_multiple_pointers.dart
bool areThereDuplicates<T extends Comparable<T>>(List<T> args) {
  args.sort();

  int left = 0;
  int right = 1;

  while (right < args.length) {
    if (args[left] == args[right]) {
      return true;
    }

    left++;
    right++;
  }

  return false;
}

void main() {
  print(areThereDuplicates<String>(['a', 'b', 'c', 'a']));
}
